tests/tools/onert_train/src/args.h

```diff
@@ -81,11 +89,11 @@ class Args
   std::string _load_raw_expected_filename;
   bool _mem_poll;
   int _epoch;
-  int _batch_size;
-  float _learning_rate;
-  int _loss_type;
-  int _loss_reduction_type;
-  int _optimizer_type;
+  std::optional<int> _batch_size;
+  std::optional<float> _learning_rate;
+  std::optional<NNFW_TRAIN_LOSS> _loss_type;
+  std::optional<NNFW_TRAIN_LOSS_REDUCTION> _loss_reduction_type;
+  std::optional<NNFW_TRAIN_OPTIMIZER> _optimizer_type;
```

Comment:
@jyoungyun 

I'd like to hear your opinion about this change.

#### AS-IS
* `onert_train/args.h` holds integer value and parse it to enum `NNFW_TRAIN_*` in `onert_train`'s main function.

#### TO-BE
* `onert_train/args.h` directly holds enum `NNFW_TRAIN_*`.
* Since it holds an enum value, to distinguish the not-given value use `optional` class.
 

Since boost has a problem with c++17, I'm trying to find another way,  But before going further, I'd like to hear your opinion about the overall direction.